core/templates/components/question-directives/questions-list/questions-list.component.ts

```diff
@@ -13,650 +13,677 @@
+
+interface GroupedSkillSummaries {
+  current: SkillSummaryBackendDict[];
+  others: SkillSummaryBackendDict[];
+}
+
+@Component({
+  selector: 'oppia-questions-list',
+  templateUrl: './questions-list.component.html'
+})
+export class QuestionsListComponent implements OnInit, OnDestroy, OnChanges {
+  @Input() allSkillSummaries: ShortSkillSummary[];
+  @Input() canEditQuestion: boolean;
+  @Input() groupedSkillSummaries: GroupedSkillSummaries;
+  @Input() selectedSkillId: string;
+  @Input() selectSkillModalIsShown: boolean;
+  @Input() skillIdToRubricsObject: Record<string, Rubric>;
+  @Input() skillsCategorizedByTopics: CategorizedSkills;
+  @Input() untriagedSkillSummaries: SkillSummary[];
+  @Input() skillDescriptionsAreShown: boolean;
+
+  associatedSkillSummaries: ShortSkillSummary[];
+  deletedQuestionIds: string[];
+  difficulty: number;
+  difficultyCardIsShown: boolean;
+  editorIsOpen: boolean;
+  isSkillDifficultyChanged: boolean;
+  linkedSkillsWithDifficulty: SkillDifficulty[];
+  misconceptionIdsForSelectedSkill: unknown[];
+  misconceptionsBySkill: MisconceptionSkillMap;
+  newQuestionIsBeingCreated: boolean;
+  newQuestionSkillDifficulties: number[] | number;
+  newQuestionSkillIds: string[];
+  question: Question;
+  questionEditorIsShown: boolean;
+  questionId: string;
+  questionIsBeingSaved: boolean;
+  questionIsBeingUpdated: boolean;
+  questionStateData: State;
+  questionSummariesForOneSkill: QuestionSummaryForOneSkill[] = [];
+  showDifficultyChoices: boolean;
+  skillLinkageModificationsArray: SkillLinkageModificationsArray[];
+  directiveSubscriptions = new Subscription();
+
+  constructor(
+    private alertsService: AlertsService,
+    private changeDetectorRef: ChangeDetectorRef,
+    private contextService: ContextService,
+    private editableQuestionBackendApiService:
+      EditableQuestionBackendApiService,
+    private focusManagerService: FocusManagerService,
+    private imageLocalStorageService: ImageLocalStorageService,
+    private misconceptionObjectFactory: MisconceptionObjectFactory,
+    private ngbModal: NgbModal,
+    private questionObjectFactory: QuestionObjectFactory,
+    private questionsListService: QuestionsListService,
+    private questionUndoRedoService: QuestionUndoRedoService,
+    private questionValidationService: QuestionValidationService,
+    private skillBackendApiService: SkillBackendApiService,
+    private skillEditorRoutingService: SkillEditorRoutingService,
+    private utilsService: UtilsService,
+    private windowDimensionsService: WindowDimensionsService,
+    private windowRef: WindowRef,
+  ) { }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    // QuestionsListService takes time to
+    // update QuestionSummariesForOneSkill when new
+    // skill is selected from dropdown
+    // in topic quesiton tab component.
+    setTimeout(() => {
+      this.getQuestionSummariesForOneSkill();
+      this.changeDetectorRef.detectChanges();
+    }, 300);
```

Comment:
Use a EventEmitter to resolve this.

Reply:
earlier we are using this because when new skill is selected from drop downtown in topic question tab components. it takes time to update view.

this code  has been updated by subscribing to  `onQuestionSummariesInitialized from questionsListService` and calling 
getQuestionSummariesForOneSkill() + changeDetectorRef.detectChanges to update view in it. `now when user will select another skill in dropdown it will automatically update view`.

#### Proof that changes are correct
[Screencast from 22-07-22 11:26:00 PM IST.webm](https://user-images.githubusercontent.com/77006897/180497392-26e3eb04-e626-4365-8b99-62e2d2d53c4c.webm)